examples/menus/index.ts

```diff
@@ -0,0 +1,358 @@
+        shortcut: 'Ctrl+F',
+        handler: logItem
+      },
+      {
+        text: 'Find Next',
+        shortcut: 'F3',
+        handler: logItem
+      },
+      {
+        text: 'Find Previous',
+        shortcut: 'Shift+F3',
+        handler: logItem
+      },
+      {
+        type: 'separator'
+      },
+      {
+        text: 'Replace...',
+        shortcut: 'Ctrl+H',
+        handler: logItem
+      },
+      {
+        text: 'Replace Next',
+        shortcut: 'Ctrl+Shift+H',
+        handler: logItem
+      }
+    ]
+  },
+  {
+    text: 'View',
+    disabled: true
+  },
+  {
+    type: 'separator'
+  },
+  {
+    text: 'Help',
+    submenu: [
+      {
+        text: 'Documentation',
+        handler: logItem
+      },
+      {
+        text: 'About',
+        handler: logItem
+      }
+    ]
+  }
+];
+
+
+var CONTEXT_MENU_TEMPLATE = [
+  {
+    text: '&Copy',
+    shortcut: 'Ctrl+C',
+    className: 'copy',
+    handler: logItem
+  },
+  {
+    text: 'Cu&t',
+    shortcut: 'Ctrl+X',
+    className: 'cut',
+    handler: logItem
+  },
+  {
+    text: '&Paste',
+    shortcut: 'Ctrl+V',
+    className: 'paste',
+    handler: logItem
+  },
+  {
+    type: 'separator'
+  },
+  {
+    text: '&New Tab',
+    handler: logItem
+  },
+  {
+    text: '&Close Tab',
+    handler: logItem
+  },
+  {
+    type: 'check',
+    checked: true,
+    text: '&Save On Exit',
+    handler: (item: MenuItem) => {
+      item.checked = !item.checked;
+      console.log('Save On Exit:', item.checked);
+    }
+  },
+  {
+    type: 'separator'
+  },
+  {
+    text: 'Task Manager',
+    disabled: true
+  },
+  {
+    type: 'separator'
+  },
+  {
+    text: 'More...',
+    submenu: [
+      {
+        text: 'One',
+        handler: logItem
+      },
+      {
+        text: 'Two',
+        handler: logItem
+      },
+      {
+        text: 'Three',
+        handler: logItem
+      },
+      {
+        text: 'Four',
+        handler: logItem
+      }
+    ]
+  },
+  {
+    type: 'separator'
+  },
+  {
+    text: 'Close',
+    className: 'close',
+    handler: logItem
+  }
+];
+
+
+function main(): void {
+
+  var menuBar = MenuBar.fromTemplate(MENU_BAR_TEMPLATE);
+  var contextMenu = Menu.fromTemplate(CONTEXT_MENU_TEMPLATE);
+
+  document.addEventListener('contextmenu', (event: MouseEvent) => {
+    event.preventDefault();
+    var x = event.clientX;
+    var y = event.clientY;
+    contextMenu.popup(x, y);
+  });
+
+  TabPanel.setTab(menuBar, new Tab('Menu'));
+
+  var cmSource = new CodeMirrorWidget({
+    mode: 'text/javascript',
+    lineNumbers: true,
+    tabSize: 2,
+  });
+  cmSource.loadTarget('./index.ts');
+
+  var cmCss = new CodeMirrorWidget({
+    mode: 'text/css',
+    lineNumbers: true,
+    tabSize: 2,
+  });
+  cmCss.loadTarget('./index.css');
+
+  TabPanel.setTab(cmSource, new Tab('Source'));
+  TabPanel.setTab(cmCss, new Tab('CSS'));
+
+  var panel = new TabPanel();
+  panel.id = 'main';
+  panel.widgets = [menuBar, cmSource, cmCss];
```

Comment:
A couple things:

1) The menu bar should not be added directly the tab panel. It should be added as the child of another vanilla widget, and that widget should be added to the tab panel. This will allow the menubar to layout naturally, as intended.

2) The context menu should probably only popup for the demo panel. So the context menu listener should probably be attached to the widget which hosts the menu bar, instead of the document body.